webservices/common/views.py

```diff
@@ -148,9 +148,67 @@ def join_sub_queries(self, kwargs, primary_field, secondary_field=None):
         query = query.options(*self.query_options)
         return query
 
+    def join_year_queries(self, kwargs):
+        """Build and compose per-field subqueries using `UNION ALL`.
+        If all years, break up two_year_transaction period into 10-year ranges
+        starting in 2020
+        """
+        queries = []
+        temp_kwargs = {}
+        if kwargs.get("two_year_transaction_period") == self.ALL_YEARS:
+            # Remove kwargs so we don't filter twice
+            temp_kwargs = copy.deepcopy(kwargs)
+            del temp_kwargs["two_year_transaction_period"]
+            # TODO: ^ is this overkill?
+            # Break into 10-year ranges starting in 2000
+            for min_year, max_year in self.get_decade_range(self.ALL_YEARS):
+                query = self.build_union_range_subquery(temp_kwargs, min_year, max_year)
+                queries.append(query.subquery().select())
+        else:
+            for argument in kwargs.get("two_year_transaction_period", []):
+                temp_kwargs["two_year_transaction_period"] = [argument]
+                query = self.build_union_subquery(kwargs, temp_kwargs)
+                queries.append(query.subquery().select())
+        query = models.db.session.query(
+            self.model
+        ).select_entity_from(
+            sa.union_all(*queries)
+        )
+        query = query.options(*self.query_options)
+        return query
+
+# TODO: Add automated testing for this
+    def get_decade_range(self, year_range):
```

Comment:
Why not just start with 2000 and add to it up to the current year?

Reply:
This isn't quite right yet. Add test coverage before trying to "simplify" since version 1 seems to work.

```
def get_decade_range(start_year):
    """Get a list of tuples representing min, max years"""
    decade_range = [(None, start_year)]
    # Start with everything up to the current year
    for i in range (start_year, 2022, 10):
        decade_range.append((i, i+10))
    decade_range.append((i+10, None))
    return decade_range
```